File: 백준/Silver/21921. 블로그/블로그.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

	static int N, X;
	static int[] arr;

	public static void main(String[] args) throws NumberFormatException, IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		N = Integer.parseInt(st.nextToken());
		X = Integer.parseInt(st.nextToken());

		arr = new int[N];
		st = new StringTokenizer(br.readLine());
		for (int i = 0; i < N; i++) {
			arr[i] = Integer.parseInt(st.nextToken());
		}

		int maxVisitor = 0;
		int maxCount = 1;
		for (int x = 0; x < X; x++) {
			maxVisitor += arr[x];
		}

		int sum = maxVisitor;
		for (int i = 0; i < N - X; i++) {
			int visitor = sum;
			visitor -= arr[i];
			visitor += arr[i + X];
			if (maxVisitor < visitor) {
				maxVisitor = visitor;
				maxCount = 1;
			} else if (maxVisitor == visitor) {
				maxCount++;
			}
			sum = visitor;
		}

		if (maxVisitor == 0) {
			System.out.println("SAD");
		} else {
			System.out.println(maxVisitor);
			System.out.println(maxCount);
		}
	}

}

File: 백준/Silver/21921. 블로그/블로그.py
import sys
input = sys.stdin.readline


N, X = map(int, input().split())
arr = list(map(int, input().split()))

max_visitor = 0
max_count = 1
for x in range(X):
    max_visitor += arr[x]

ssum = max_visitor
for i in range(N-X):
    visitor = ssum
    visitor -= arr[i]
    visitor += arr[i+X]
    if (max_visitor < visitor):
        max_visitor = visitor
        max_count = 1
    elif max_visitor == visitor:
        max_count += 1
    ssum = visitor

if max_visitor == 0:
    print("SAD")
else:
    print(max_visitor)
    print(max_count)